lib/mpl_toolkits/mplot3d/tests/test_art3d.py

```diff
@@ -54,3 +54,32 @@ def test_zordered_error():
     ax.add_collection(Line3DCollection(lc))
     ax.scatter(*pc, visible=False)
     plt.draw()
+
+
+def test_generate_normals():
+
+    # Following code is an example taken from
+    # https://stackoverflow.com/questions/18897786/transparency-for-poly3dcollection-plot-in-matplotlib
+    # and modified to test _generate_normals function
```

Comment:
To the other devs, it's not clear to me whether we can include stack overflow code directly like this? It's not clear to me if the CC BY-SA license for stack overflow posts is compatible with matplotlib's license.

Reply:
I think a better test is the one in the issue, with a note that it is a smoke test and a reference to the issue. The added test is not really testing any rendering aspects anyway.

Reply:
Better in the sense that we do not have to worry about the SE-connection and that the SE-test doesn't really add anything more, just more complicated code.